src/clientHandler.ts

```diff
@@ -1,168 +1,53 @@
 
-    client.onDidChangeState((event) => {
-      console.log(`Client: ${State[event.oldState]} --> ${State[event.newState]}`);
-      if (event.newState === State.Stopped) {
-        this.reporter.sendTelemetryEvent('stopClient');
-      }
-    });
-
-    return client;
-  }
-
-  private async getServerOptions(): Promise<ServerOptions> {
-    const cmd = await this.lsPath.resolvedPathToBinary();
-    const serverArgs = config('terraform').get<string[]>('languageServer.args', []);
-    const executable: Executable = {
-      command: cmd,
-      args: serverArgs,
-      options: {},
-    };
-    const serverOptions: ServerOptions = {
-      run: executable,
-      debug: executable,
-    };
-    this.outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')}`);
-    return serverOptions;
-  }
-
-  private getInitializationOptions() {
-    const rootModulePaths = config('terraform-ls').get<string[]>('rootModules', []);
-    const terraformExecPath = config('terraform-ls').get<string>('terraformExecPath', '');
-    const terraformExecTimeout = config('terraform-ls').get<string>('terraformExecTimeout', '');
-    const terraformLogFilePath = config('terraform-ls').get<string>('terraformLogFilePath', '');
-    const excludeModulePaths = config('terraform-ls').get<string[]>('excludeRootModules', []);
-    const ignoreDirectoryNames = config('terraform-ls').get<string[]>('ignoreDirectoryNames', []);
-
-    const ignoreSingleFileWarning = config('terraform').get<boolean>('languageServer.ignoreSingleFileWarning', false);
-
-    if (rootModulePaths.length > 0 && excludeModulePaths.length > 0) {
-      throw new Error(
-        'Only one of rootModules and excludeRootModules can be set at the same time, please remove the conflicting config and reload',
-      );
-    }
-
-    const experimentalFeatures = config('terraform-ls').get('experimentalFeatures');
-    const initializationOptions = {
-      experimentalFeatures,
-      ignoreSingleFileWarning,
-      ...(terraformExecPath.length > 0 && { terraformExecPath }),
-      ...(terraformExecTimeout.length > 0 && { terraformExecTimeout }),
-      ...(terraformLogFilePath.length > 0 && { terraformLogFilePath }),
-      ...(rootModulePaths.length > 0 && { rootModulePaths }),
-      ...(excludeModulePaths.length > 0 && { excludeModulePaths }),
-      ...(ignoreDirectoryNames.length > 0 && { ignoreDirectoryNames }),
-    };
-    return initializationOptions;
-  }
-
-  public async stopClient(): Promise<void> {
-    if (this.client === undefined) {
-      return;
-    }
-
-    await this.client.stop();
-    console.log('Client stopped');
-  }
-
-  public getClient(): LanguageClient | undefined {
-    return this.client;
-  }
+  return serverOptions;
+}
 
-  public clientSupportsCommand(cmdName: string): boolean {
-    return this.commands.includes(cmdName);
-  }
+export function getInitializationOptions() {
```

Comment:
This is basically a set of settings masquerading as a class method. My intention here is to make room for this to be added to a configuration builder when we tackle #791